CL/cl_icd_layer.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020 The Khronos Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * OpenCL is a trademark of Apple Inc. used under license by Khronos.
+ */
+
+#ifndef __ICD_LAYER_H
```

Comment:
This is fairly minor, but since we have no backward compatibility concerns (yet), I think we should use an include guard that is not technically a reserved identifier (see https://github.com/KhronosGroup/OpenCL-Headers/issues/124).  Could we use `OPENCL_CL_ICD_LAYER_H` instead?  This seems to be the informal convention adopted by `cl_half.h` and `cl_icd.h`.

Reply:
That's a very good suggestion Ben, thanks for catching this.